pom.xml

```diff
@@ -546,8 +546,35 @@
           <artifactId>maven-checkstyle-plugin</artifactId>
           <version>3.1.1</version>
             <configuration>
+              <failOnViolation>true</failOnViolation>
+              <includeTestSourceDirectory>true</includeTestSourceDirectory>
+              <sourceDirectories>
+                  <directory>${project.basedir}/src/main/java</directory>
+                  <directory>${project.basedir}/src/main/scala</directory>
+              </sourceDirectories>
+              <testSourceDirectories>
+                  <directory>${project.basedir}/src/test/java</directory>
+              </testSourceDirectories>
               <configLocation>dev/checkstyle.xml</configLocation>
```

Comment:
does this option take effect for all sub-modules?

Reply:
Yes, in the CI
```
[INFO] 
[INFO] --- maven-checkstyle-plugin:3.1.1:check (default) @ spark-sql-columnar-shims-spark32 ---
[INFO] You have 0 Checkstyle violations.
```

Reply:
That's great. Thanks.